### PR TITLE
Rack intercom api

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- ~~Adds the following functionality:~~
  - ~~Enable/Disable intercoms in a vehicle (to simulate damage).~~
  - ~~Enable/Disable racks in a vehicle (to simulate damage).~~
- Adds the following API functions:
  - ~~`acre_api_fnc_areIntercomsInVehicleDisabled`: checks whether the intercom is disabled~~
  - ~~`acre_api_fnc_disableIntercomsInVehicle`: disables intercom systems in a vehicle~~
  - ~~`acre_api_fnc_enableIntercomsInVehicle`: enables intercom systems in a vehicle~~
  - `acre_api_fnc_initVehicleRacks`: initialises the racks in a vehicle.
  - ~~`acre_api_fnc_areRacksInVehicleDisabled`: checks whether racks are disabled~~
  - ~~`acre_api_fnc_enableRacksInVehicle`; enables racks in a vehicle~~
  - ~~`acre_api_fnc_disableRacksInVehicle`; disables racks in a vehicle~~
  - `acre_api_fnc_getMountedRackRadio`: returns the mounted rack radio
  - `acre_api_fnc_getVehicleRacks`; returns vehicle racks
  - `acre_api_fnc_mountRackRadio`: mounts a radio.
  - `acre_api_fnc_unmountRackRadio`:  dismounts a radio.
- Fixes some bugs

Enable/disable intercoms and racks in a vehicle will not be ready for 2.5.0. I was not happy with it since it was a on/off feature and I would prefer to have a progressive effect. Maybe noise?

### CODE REVIEW

File: addons/api/fnc_initVehicleRacks.sqf [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/*
+ * Author: ACRE2Team
+ * Initialises all racks in the vehicle.
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ *
+ * Return Value:
+ * Setup succesful <BOOL>
```
Comment (jonpas) [inline addons/api/fnc_initVehicleRacks.sqf]: `successful`

File: addons/api/fnc_mountRackRadio.sqf [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Author: ACRE2Team
+ * Sets the given radio as mounted
```
Comment (jonpas) [inline addons/api/fnc_mountRackRadio.sqf]: Missing final dot.

File: addons/api/fnc_mountRackRadio.sqf [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * Arguments:
+ * 0: Rack ID <STRING>
+ * 1: Radio to mount <STRING>
+ * 2: Unit with radio to mount <OBJECT>
+ *
+ * Return Value:
+ * Setup succesful <BOOL>
```
Comment (jonpas) [inline addons/api/fnc_mountRackRadio.sqf]: `successful`

File: addons/api/fnc_unmountRackRadio.sqf [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ * Removes a mounted radio
+ *
+ * Arguments:
+ * 0: Rack ID <STRING>
+ * 1: Radio to unmount <STRING>
+ * 2: Unit with radio to unmount <OBJECT>
+ * 3: Remove radio after unmounting <BOOL><OPTIONAL>
+ *
+ * Return Value:
+ * Setup succesful <BOOL>
```
Comment (jonpas) [inline addons/api/fnc_unmountRackRadio.sqf]: `successful`

File: addons/api/fnc_unmountRackRadio.sqf [hunk 5]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Author: ACRE2Team
+ * Removes a mounted radio
+ *
+ * Arguments:
+ * 0: Rack ID <STRING>
+ * 1: Radio to unmount <STRING>
+ * 2: Unit with radio to unmount <OBJECT>
+ * 3: Remove radio after unmounting <BOOL><OPTIONAL>
```
Comment (jonpas) [inline addons/api/fnc_unmountRackRadio.sqf]: ` * 3: Remove radio after unmounting <BOOL> (default: false)`

File: addons/api/fnc_unmountRackRadio.sqf [hunk 6]
```diff
@@ -0,0 +1,78 @@
+/*
+ * Author: ACRE2Team
+ * Removes a mounted radio
```
Comment (jonpas) [inline addons/api/fnc_unmountRackRadio.sqf]: Missing final dot.

**Review by jonpas — CHANGES_REQUESTED** [review]

File: addons/api/fnc_initVehicleRacks.sqf (hunk 1 above)
Comment (TheMagnetar) [reply addons/api/fnc_initVehicleRacks.sqf]: This is common in most of API functions.

Comment (jonpas) [reply addons/api/fnc_initVehicleRacks.sqf]: Well it's wrong AFAIK.

**Review by jonpas — APPROVED** [review]

**Pull request merged**